File: Lab 2/Alex/consumer.py
from kafka import KafkaConsumer

TOPIC="lab2"

consumer = KafkaConsumer(bootstrap_servers="0.0.0.0:9094")

topics = []
topics.append(TOPIC)
consumer.subscribe(topics=topics)

for message in consumer:
    message = message.value.decode()
    print(message)

consumer.close()

File: Lab 2/Alex/consumer.js
const {Kafka} = require("kafkajs")
const topic = "lab2"

run();
async function run(){
    try
    {
        const kafka = new Kafka({
            "brokers" :["localhost:9093"]
        })

        const consumer = kafka.consumer({"groupId": "test"})
        console.log("Connecting.....")
        await consumer.connect()
        console.log("Connected!")

        await consumer.subscribe({
            "topic": topic,
            "fromBeginning": true
        })

        await consumer.run({
            "eachMessage": async result => {
                console.log(`RVD Msg ${result.message.value} on partition ${result.partition}`)
            }
        })


    }
    catch(ex)
    {
        console.error(`Something bad happened ${ex}`)
    }
    finally{

    }


}